### ISSUE TITLE
TypedDict method gotchas

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#1022]
**Describe the bug**
When attempting to use `setdefault`, `pop` or `__delitem__` (but not `del` which I assume bypasses the method) on a `TypedDict`, Pyright reports that the argument cannot be assigned to a parameter of type `NoReturn`.  For `get`, Pyright reports that the type is partially unknown.  For `TypedDict`s with a totality value of false, key access is not conditionalised. 

**To Reproduce**
Type check the snippet below.

**Expected behavior**
That `pop` and `__delitem__` would not produce any errors on non-total `TypedDict`s, and that `get` and `setdefault` would work in all cases.  ~~That non-total `TypedDict`s would not be subscriptable without a guard, e.g. `'bar' in foo and foo['bar']`.~~ The PEP recommends the opposite.

**Screenshots or Code**
If applicable, add screenshots or the text of the code (surrounded by triple back ticks) to help explain your problem.
```python
# pyright: strict

from typing import TypedDict

class Foo(TypedDict, total=False):
    bar: str

foo: Foo = {}

# Ok but might `KeyError` if `total=False`
foo['bar']  

# Type of "get" is partially unknown
#   Type of "get" is "Overload[(k: Literal['bar'], default: str) -> str, (k: str, default: Unknown) -> Unknown]"
foo.get('bar', '')

# Argument of type "Literal['bar']" cannot be assigned to parameter "k" of type "NoReturn" in function "setdefault"
#   "Literal['bar']" is incompatible with "NoReturn"
foo.setdefault('bar', '1')

# Argument of type "Literal['bar']" cannot be assigned to parameter "k" of type "NoReturn" in function "pop"
#   "Literal['bar']" is incompatible with "NoReturn"
foo.pop('bar')

# Argument of type "Literal['bar']" cannot be assigned to parameter "k" of type "NoReturn" in function "__delitem__"
#   "Literal['bar']" is incompatible with "NoReturn"
foo.__delitem__('bar')

```

**VS Code extension or command-line**
Extension, 1.1.69

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is now addressed in Pyright 1.1.71, which I just published.
